agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryClient.java

```diff
@@ -193,50 +200,69 @@ public void trackStatsbeatAsync(TelemetryItem telemetry) {
   }
 
   public CompletableResultCode flushChannelBatcher() {
-    if (channelBatcher != null) {
-      return channelBatcher.forceFlush();
+    if (generalChannelBatcher != null) {
+      return generalChannelBatcher.forceFlush();
     } else {
       return CompletableResultCode.ofSuccess();
     }
   }
 
-  public BatchSpanProcessor getChannelBatcher() {
-    if (channelBatcher == null) {
+  private BatchSpanProcessor getGeneralChannelBatcher() {
+    if (generalChannelBatcher == null) {
       synchronized (channelInitLock) {
-        if (channelBatcher == null) {
-          File telemetryFolder;
-          LocalFileLoader localFileLoader = null;
-          LocalFileWriter localFileWriter = null;
-          if (!readOnlyFileSystem) {
-            telemetryFolder = LocalStorageUtils.getOfflineTelemetryFolder();
-            LocalFileCache localFileCache = new LocalFileCache(telemetryFolder);
-            localFileLoader =
-                new LocalFileLoader(
-                    localFileCache, telemetryFolder, statsbeatModule.getNonessentialStatsbeat());
-            localFileWriter =
-                new LocalFileWriter(
-                    localFileCache, telemetryFolder, statsbeatModule.getNonessentialStatsbeat());
-          }
-
-          TelemetryChannel channel =
-              TelemetryChannel.create(
-                  endpointProvider.getIngestionEndpointUrl(),
-                  localFileWriter,
-                  ikeyEndpointMap,
-                  statsbeatModule,
-                  false,
-                  aadAuthentication);
-
-          if (!readOnlyFileSystem) {
-            LocalFileSender.start(localFileLoader, channel);
-          }
+        if (generalChannelBatcher == null) {
+          generalChannelBatcher = initChannelBatcher(generalExportQueueCapacity, 512, "general");
```

Comment:
curios why the queue size if 512 here? I thought the default size is 2048?

Reply:
512 is the max batch size not the queue size